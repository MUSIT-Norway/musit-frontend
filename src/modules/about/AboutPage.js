import React from 'react';
import { Grid, Row } from 'react-bootstrap';
import './AboutPage.css';
import { I18n } from 'react-i18nify';
import NorwegianTranslation from './AboutPage_no.html.jsx';
import EnglishTranslation from './AboutPage_en.html.jsx';
import Logos from '../../components/logos/Logos';
import inject from 'react-rxjs/dist/RxInject';

export const AboutPage = (props) => {
  const Translated = props.getLocale() === 'no' ? NorwegianTranslation : EnglishTranslation;
  return (
    <div>
      <main>
        <Grid>
          <Row className="row-centered">
            <div className="aboutPanel">
              <div>
                <Translated {...props} />
                <Logos />
              </div>
            </div>
          </Row>
        </Grid>
      </main>
    </div>
  );
};

export default inject({}, {}, { getLocale: () => I18n._locale })(AboutPage);