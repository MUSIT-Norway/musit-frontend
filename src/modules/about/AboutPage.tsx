import * as React from 'react';
import { Grid, Row } from 'react-bootstrap';
import './AboutPage.css';
import { I18n } from 'react-i18nify';
import NorwegianTranslation from './AboutPage_no.html';
import EnglishTranslation from './AboutPage_en.html';
import Logos from '../../components/logos/Logos';
import { RxInjectLegacy as inject } from '../../shared/react-rxjs-patch/';
import { TODO } from '../../types/common';

export const AboutPage = (props:TODO) => {
  const Translated =
    props.getLocale() === 'no' ? NorwegianTranslation : EnglishTranslation;
  return (
    <div>
      <main>
        <Grid>
          <Row className="row-centered">
            <div className="aboutPanel">
              <div>
                <Translated {...props} />
                <Logos />
              </div>
            </div>
          </Row>
        </Grid>
      </main>
    </div>
  );
};

export default inject({}, {}, { getLocale: () => I18n._locale })(AboutPage);
