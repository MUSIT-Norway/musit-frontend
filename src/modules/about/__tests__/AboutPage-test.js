import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import React from 'react';
import { AboutPage } from '../AboutPage';
import AboutPageEn from '../AboutPage_en.html';
import AboutPageNo from '../AboutPage_no.html';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({ adapter: new Adapter() });

describe('AboutPage', () => {
  it('should render norwegian text', () => {
    const wrapper = shallow(<AboutPage getLocale={() => 'no'} />);
    expect(wrapper.find(AboutPageNo).length).toBe(1);
    expect(wrapper.find(AboutPageEn).length).toBe(0);
  });
  it('should render english text', () => {
    const wrapper = shallow(<AboutPage getLocale={() => 'en'} />);
    expect(wrapper.find(AboutPageEn).length).toBe(1);
    expect(wrapper.find(AboutPageNo).length).toBe(0);
  });

  it('should have norwegian text', () => {
    const wrapper = shallow(<AboutPageNo />);
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });

  it('should have english text', () => {
    const wrapper = shallow(<AboutPageEn />);
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
