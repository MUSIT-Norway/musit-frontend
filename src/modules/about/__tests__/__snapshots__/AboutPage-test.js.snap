exports[`AboutPage should have english text 1`] = `
<div>
  <h2>
    MUSIT - The University museums IT-organisation
  </h2>
  <br />
  <br />
  <p>
    The University museums IT-organisation (MUSIT) is a national collaboration between the University of Bergen, the University of Oslo, University of Stavanger, Norwegian University of Science and Technology in Trondheim , and the University of Tromsø -Norway\'s Arctic University.
  </p>
  <p>
    The collaboration is between the University Museum of Bergen, Archeological Museum in Stavanger, Museum of Cultural History in Oslo, Natural History Museum in Oslo, NTNU University museum in Trondheim, and the Tromsø University museum.
  </p>
  <p>
    These university museums are custodians of Norway’s largest natural and cultural history collections encompassing a number of disciplines such as archeology, botany, ethnography, geology, numismatic and zoology. MUSITs core purpose is to ensure the operation, maintenance and development of the university museums joint collection databases in order to enable the dissemination of these data-rich collections for use in research, conservation policy and management, and education.
  </p>
  <br />
  <br />
</div>
`;

exports[`AboutPage should have norwegian text 1`] = `
<div>
  <h2>
    MUSIT - Universitetsmuseenes IT-organisasjon
  </h2>
  <br />
  <br />
  <p>
    Universitetsmuseenes IT-organisasjon (MUSIT) er et nasjonalt samarbeidstiltak mellom Universitetet i Bergen, Universitetet i Oslo, Universitetet i Stavanger, Norges teknisk-naturvitenskapelige universitet og UiT Norges arktiske universitet.
  </p>
  <p>
    Samarbeidstiltaket gjelder Arkeologisk museum, Kulturhistorisk museum, Naturhistorisk museum, NTNU Vitenskapsmuseet, Tromsø museum – Universitetsmuseet, og Universitetsmuseet i Bergen.
  </p>
  <p>
    Universitetsmuseene forvalter de største samlingene i Norge innenfor en rekke fagfelt som arkeologi, botanikk, etnografi, geologi, numismatikk og zoologi. MUSIT er den sentrale forvalter av universitetsmuseenes samlingsdata, og skal tilby løsninger som gjør datasett lett tilgjengelig for bruk i forskning, forvaltning og formidling.
  </p>
  <br />
  <br />
</div>
`;
